state/watcher/hubwatcher.go

```diff
@@ -436,6 +458,9 @@ func (w *HubWatcher) flush() bool {
 			case e.ch <- outChange:
 				w.logger.Tracef("%p e.ch=%v has been notified %v", w, e.ch, outChange)
 				watchersNotified = true
+			case <-deadline:
+				w.logger.Errorf("%p e.ch=%v did not accept %v - missing Unwatch?\nwatch source:\n%s", w, e.ch, outChange, e.watchSource)
```

Comment:
There is a bit of a weird race here, but not one that I think we care about. It is possible that we may iterate through the inner loop more than once and have the underlying sync event change due to the processing of `w.handle` request. I don't think this will every actually be a problem due to the size of the deadline, and the number of requests. All this really means is that we may inaccurately give a particular item 9.999s rather than 10s, so, no big deal.

Critical maybe? In similar areas elsewhere, I've prefixed the message with "programming error" as it isn't an expected error.

Reply:
Yeah, I'll change it to critical and indicate that it's a programming error.